panic: assignment to entry in nil map

Hi !

Love what you have done here, followed your clear instructions but regardless of whether I use ES or LS, I still get the nil map error.

Ubuntu/Nvidia P40s. metricbeats is working on the same box at the same time without difficulty.

Activity:

Thu Aug 23 18:23:32 2018
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.130                Driver Version: 384.130                   |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P40           Off  | 00000000:3B:00.0 Off |                    0 |
| N/A   63C    P0   183W / 250W |  21939MiB / 22912MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P40           Off  | 00000000:D8:00.0 Off |                    0 |
| N/A   66C    P0   182W / 250W |  21939MiB / 22912MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0    167989      C   python3                                    21927MiB |
|    1    167989      C   python3                                    21927MiB |
+-----------------------------------------------------------------------------+


./nvidiagpubeat -c nvidiagpubeat.yml -e -d "*"
2018/08/23 22:20:51.237871 beat.go:267: INFO Home path: [/root/beats_dev/src/github.com/deepujain/nvidiagpubeat] Config path: [/root/beats_dev/src/github.com/deepujain/nvidiagpubeat] Data path: [/root/beats_dev/src/github.com/deepujain/nvidiagpubeat/data] Logs path: [/root/beats_dev/src/github.com/deepujain/nvidiagpubeat/logs]
2018/08/23 22:20:51.237886 beat.go:177: INFO Setup Beat: nvidiagpubeat; Version: 5.2.3
2018/08/23 22:20:51.237899 processor.go:43: DBG  Processors: include_fields=cpu, @timestamp, type
2018/08/23 22:20:51.237903 beat.go:183: DBG  Initializing output plugins
2018/08/23 22:20:51.237932 logstash.go:90: INFO Max Retries set to: 3
2018/08/23 22:20:51.237972 logp.go:219: INFO Metrics logging every 30s
2018/08/23 22:20:51.237973 outputs.go:106: INFO Activated logstash as output plugin.
2018/08/23 22:20:51.238011 publish.go:234: DBG  Create output worker
2018/08/23 22:20:51.238066 publish.go:276: DBG  No output is defined to store the topology. The server fields might not be filled.
2018/08/23 22:20:51.238088 publish.go:291: INFO Publisher name: GROOT01
2018/08/23 22:20:51.238904 async.go:63: INFO Flush Interval set to: 1s
2018/08/23 22:20:51.238915 async.go:64: INFO Max Bulk Size set to: 2048
2018/08/23 22:20:51.238920 async.go:72: DBG  create bulk processing worker (interval=1s, bulk size=2048)
2018/08/23 22:20:51.239001 beat.go:207: INFO nvidiagpubeat start running.
2018/08/23 22:20:51.239017 nvidiagpubeat.go:38: INFO nvidiagpubeat is running for ** production ** environment. ! Hit CTRL-C to stop it.
2018/08/23 22:20:52.278616 nvidiagpubeat.go:55: INFO Event generated, Attempting to publish to configured output.
2018/08/23 22:20:52.278752 client.go:184: DBG  Publish: {
  "@timestamp": "2018-08-23T22:20:52.277Z"
}
2018/08/23 22:20:52.278816 client.go:184: DBG  Publish: {
  "@timestamp": "2018-08-23T22:20:52.277Z"
}
2018/08/23 22:20:52.278859 logp.go:245: INFO Total non-zero values:  libbeat.publisher.messages_in_worker_queues=2 libbeat.publisher.published_events=2
2018/08/23 22:20:52.278873 logp.go:246: INFO Uptime: 1.046453186s
2018/08/23 22:20:52.278877 asm_amd64.s:515: INFO nvidiagpubeat stopped.
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*client).annotateEvent(0xc42015fa80, 0x0)
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/client.go:150 +0xe0
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*client).PublishEvent(0xc42015fa80, 0x0, 0x0, 0x0, 0x0, 0x1)
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/client.go:99 +0x4d
github.com/deepujain/nvidiagpubeat/beater.(*Nvidiagpubeat).Run(0xc42015fa40, 0xc420206000, 0xc4202060a8, 0xb)
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/beater/nvidiagpubeat.go:57 +0x375
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat.(*Beat).launch(0xc420206000, 0x9c9d98, 0x0, 0x0)
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:211 +0x706
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat.Run(0x9adebd, 0xd, 0x0, 0x0, 0x9c9d98, 0xc4200001a0, 0xc4200001a0)
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:136 +0x65
main.main()
        /root/beats_dev/src/github.com/deepujain/nvidiagpubeat/main.go:12 +0x54

Same on Centos 7.3
```
[jmacdonald@wks207 nvidiagpubeat]$ ./nvidiagpubeat -c nvidiagpubeat.yml -e -d "*"
2018/08/29 10:07:24.711499 beat.go:267: INFO Home path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat] Config path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat] Data path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/data] Logs path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/logs]
2018/08/29 10:07:24.711612 beat.go:177: INFO Setup Beat: nvidiagpubeat; Version: 5.2.3
2018/08/29 10:07:24.711706 logp.go:219: INFO Metrics logging every 30s
2018/08/29 10:07:24.711711 processor.go:43: DBG  Processors: 
2018/08/29 10:07:24.711839 beat.go:183: DBG  Initializing output plugins
2018/08/29 10:07:24.712145 output.go:167: INFO Loading template enabled. Reading template file: /home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/nvidiagpubeat.template.json
2018/08/29 10:07:24.713360 output.go:178: INFO Loading template enabled for Elasticsearch 2.x. Reading template file: /home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/nvidiagpubeat.template-es2x.json
2018/08/29 10:07:24.714429 client.go:120: INFO Elasticsearch url: http://172.16.65.122:9200
2018/08/29 10:07:24.714530 outputs.go:106: INFO Activated elasticsearch as output plugin.
2018/08/29 10:07:24.714575 publish.go:234: DBG  Create output worker
2018/08/29 10:07:24.714697 publish.go:276: DBG  No output is defined to store the topology. The server fields might not be filled.
2018/08/29 10:07:24.714763 publish.go:291: INFO Publisher name: wks207
2018/08/29 10:07:24.715508 async.go:63: INFO Flush Interval set to: 1s
2018/08/29 10:07:24.715548 async.go:64: INFO Max Bulk Size set to: 50
2018/08/29 10:07:24.715574 async.go:72: DBG  create bulk processing worker (interval=1s, bulk size=50)
2018/08/29 10:07:24.715885 beat.go:207: INFO nvidiagpubeat start running.
2018/08/29 10:07:24.715944 nvidiagpubeat.go:38: INFO nvidiagpubeat is running for ** production ** environment. ! Hit CTRL-C to stop it.
2018/08/29 10:07:25.741857 nvidiagpubeat.go:55: INFO Event generated, Attempting to publish to configured output.
2018/08/29 10:07:25.742166 client.go:184: DBG  Publish: {
  "@timestamp": "2018-08-29T10:07:25.740Z",
  "beat": {
    "hostname": "wks207",
    "name": "wks207",
    "version": "5.2.3"
  },
  "gpuIndex": 0,
  "memory": {
    "free": 4623,
    "total": 6083,
    "used": 1460
  },
  "pstate": 8,
  "temperature": {
    "gpu": 36
  },
  "utilization": {
    "gpu": 4,
    "memory": 7
  }
}
2018/08/29 10:07:25.742622 logp.go:245: INFO Total non-zero values:  libbeat.publisher.messages_in_worker_queues=1 libbeat.publisher.published_events=1
2018/08/29 10:07:25.742662 logp.go:246: INFO Uptime: 1.038700233s
2018/08/29 10:07:25.742706 asm_amd64.s:480: INFO nvidiagpubeat stopped.
panic: assignment to entry in nil map

goroutine 1 [running]:
panic(0x8badc0, 0xc42017e4d0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*client).annotateEvent(0xc420176d80, 0x0)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/client.go:150 +0xef
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*client).PublishEvent(0xc420176d80, 0x0, 0x0, 0x0, 0x0, 0x1)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/client.go:99 +0x4d
github.com/deepujain/nvidiagpubeat/beater.(*Nvidiagpubeat).Run(0xc420176d40, 0xc4200b89a0, 0xc4200b8a48, 0xb)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/beater/nvidiagpubeat.go:57 +0x38d
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat.(*Beat).launch(0xc4200b89a0, 0x993798, 0x0, 0x0)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:211 +0x699
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat.Run(0x953bdb, 0xd, 0x0, 0x0, 0x993798, 0x4efb81, 0xc4200001a0)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/beat/beat.go:136 +0x65
main.main()
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/main.go:12 +0x54

```

@deepujain 
OK, If I switch the env to "production":

```
[jmacdonald@wks207 nvidiagpubeat]$ ./nvidiagpubeat -c nvidiagpubeat.yml -e -d "*"
2018/09/12 12:36:19.301652 beat.go:267: INFO Home path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat] Config path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat] Data path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/data] Logs path: [/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/logs]
2018/09/12 12:36:19.301757 beat.go:177: INFO Setup Beat: nvidiagpubeat; Version: 5.2.3
2018/09/12 12:36:19.301827 logp.go:219: INFO Metrics logging every 30s
2018/09/12 12:36:19.301832 processor.go:43: DBG  Processors: 
2018/09/12 12:36:19.301956 beat.go:183: DBG  Initializing output plugins
2018/09/12 12:36:19.302278 output.go:167: INFO Loading template enabled. Reading template file: /home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/nvidiagpubeat.template.json
2018/09/12 12:36:19.303100 output.go:178: INFO Loading template enabled for Elasticsearch 2.x. Reading template file: /home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/nvidiagpubeat.template-es2x.json
2018/09/12 12:36:19.303798 client.go:120: INFO Elasticsearch url: http://172.16.65.122:9200
2018/09/12 12:36:19.303892 outputs.go:106: INFO Activated elasticsearch as output plugin.
2018/09/12 12:36:19.303923 publish.go:234: DBG  Create output worker
2018/09/12 12:36:19.304057 publish.go:276: DBG  No output is defined to store the topology. The server fields might not be filled.
2018/09/12 12:36:19.304129 publish.go:291: INFO Publisher name: wks207
2018/09/12 12:36:19.304923 async.go:63: INFO Flush Interval set to: 1s
2018/09/12 12:36:19.304972 async.go:64: INFO Max Bulk Size set to: 50
2018/09/12 12:36:19.305008 async.go:72: DBG  create bulk processing worker (interval=1s, bulk size=50)
2018/09/12 12:36:19.305327 beat.go:207: INFO nvidiagpubeat start running.
2018/09/12 12:36:19.305400 nvidiagpubeat.go:38: INFO nvidiagpubeat is running for ** production ** environment. ! Hit CTRL-C to stop it.
2018/09/12 12:36:20.330934 nvidiagpubeat.go:55: INFO Event generated, Attempting to publish to configured output.
2018/09/12 12:36:20.331260 client.go:184: DBG  Publish: {
  "@timestamp": "2018-09-12T12:36:20.329Z",
  "beat": {
    "hostname": "wks207",
    "name": "wks207",
    "version": "5.2.3"
  },
  "gpuIndex": 0,
  "memory": {
    "free": 4350,
    "total": 6083,
    "used": 1733
  },
  "pstate": 8,
  "temperature": {
    "gpu": 37
  },
  "utilization": {
    "gpu": 2,
    "memory": 6
  }
}
2018/09/12 12:36:21.305428 output.go:109: DBG  output worker: publish 1 events
2018/09/12 12:36:21.305563 client.go:627: DBG  ES Ping(url=http://172.16.65.122:9200, timeout=1m30s)
2018/09/12 12:36:21.309099 client.go:651: DBG  Ping status code: 200
2018/09/12 12:36:21.309146 client.go:652: INFO Connected to Elasticsearch version 6.3.2
2018/09/12 12:36:21.309193 output.go:214: INFO Trying to load template for client: http://172.16.65.122:9200
2018/09/12 12:36:21.309232 client.go:667: DBG  HEAD http://172.16.65.122:9200/_template/nvidiagpubeat  <nil>
2018/09/12 12:36:21.312482 output.go:235: INFO Template already exists and will not be overwritten.
panic: interface conversion: interface is nil, not string

goroutine 19 [running]:
panic(0x8bea60, 0xc42017a500)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.eventBulkMeta(0xbc53e0, 0xc420186140, 0x0, 0xc420174b70, 0x0, 0x96854a, 0x34)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:317 +0xd1
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.bulkEncodePublishRequest(0x7faf169020c8, 0xc4201840b8, 0xbc53e0, 0xc420186140, 0x0, 0xc4201fe000, 0x1, 0x32, 0xc42019a1f0, 0xc42018cc00, ...)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:290 +0xe0
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*Client).PublishEvents(0xc420198140, 0xc4201fe000, 0x1, 0x32, 0x0, 0x4, 0x0, 0x0, 0x0)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/client.go:235 +0x138
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents.func1(0xc42017f380)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:89 +0x7a
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).publish(0xc42017f380, 0x0, 0x0, 0x0, 0xc4201afb38, 0x5, 0x95fb53)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:144 +0x2aa
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single.(*Mode).PublishEvents(0xc42017f380, 0x0, 0x0, 0x0, 0xc4201fe000, 0x1, 0x32, 0x1, 0x1)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/mode/single/single.go:99 +0x74
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch.(*elasticsearchOutput).BulkPublish(0xc42019a180, 0x0, 0x0, 0x0, 0xc4201fe000, 0x1, 0x32, 0xc420067f88, 0xc42019e000)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/outputs/elasticsearch/output.go:308 +0x79
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).sendBulk(0xc42017c2d0, 0xc4201afce8, 0xc4201fe000, 0x1, 0x32)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:112 +0x125
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onBulk(0xc42017c2d0, 0xc4201afce8, 0xc4201fe000, 0x1, 0x32)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:88 +0x259
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*outputWorker).onMessage(0xc42017c2d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4201fe000, 0x1, 0x32)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/output.go:70 +0x9a
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).onEvent(0xc42017c2d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4201fe000, 0x1, 0x32)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:83 +0x74
github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).run(0xc42017c2d0)
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:69 +0x290
created by github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher.(*messageWorker).init
	/home/jmacdonald/Documents/code/beats_dev/src/github.com/deepujain/nvidiagpubeat/vendor/github.com/elastic/beats/libbeat/publisher/worker.go:57 +0x103

```

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,6 @@
 /.idea
 /build
-
+.venv
 .DS_Store
 /logs
 /nvidiagpubeat

--- a/nvidia/gpu.go
+++ b/nvidia/gpu.go
@@ -67,6 +67,7 @@ func (g Utilization) run(cmd *exec.Cmd, gpuCount int, query string, action Actio
 		event := common.MapStr{
 			"@timestamp": common.Time(time.Now()),
 			"gpuIndex":   gpuIndex,
+			"type":   "nvidiagpubeat",
 		}
 		for i := 0; i < len(record); i++ {
 			value, _ := strconv.Atoi(record[i])

--- a/nvidia/gpu_test.go
+++ b/nvidia/gpu_test.go
@@ -1,7 +1,6 @@
 package nvidia
 
 import (
-	"fmt"
 	"os"
 	"testing"
 )
@@ -32,11 +31,11 @@ func Test_Command_ProdEnv(t *testing.T) {
 	}
 
 	if cmd.Args[1] != "--query-gpu=myquery" {
-		t.Errorf("Expected %s, Actual %s", "--query-gpu=myquery", cmd.Args[0])
+		t.Errorf("Expected %s, Actual %s", "--query-gpu=myquery", cmd.Args[1])
 	}
 
 	if cmd.Args[2] != "--format=csv" {
-		t.Errorf("Expected %s, Actual %s", "--format=csv", cmd.Args[0])
+		t.Errorf("Expected %s, Actual %s", "--format=csv", cmd.Args[2])
 	}
 }
 
@@ -56,7 +55,7 @@ func Test_Run_ProdEnv(t *testing.T) {
 	util := NewUtilization()
 	query := "utilization.gpu,utilization.memory,memory.total,memory.free,memory.used,temperature.gpu,pstate"
 	cmd := util.command("prod", query)
-	fmt.Println(cmd.Path)
+	t.Logf("Command: %s", cmd.Path)
 	output, _ := util.run(cmd, 4, query, MockLocal{})
 
 	for _, o := range output {
@@ -64,5 +63,20 @@ func Test_Run_ProdEnv(t *testing.T) {
 			t.Errorf("output cannot be nil.")
 		}
 	}
+}
 
+func Test_Event_Contains_Type_Field(t *testing.T) {
+	util := NewUtilization()
+	query := "utilization.gpu,utilization.memory,memory.total,memory.free,memory.used,temperature.gpu,pstate"
+	cmd := util.command("prod", query)
+	t.Logf("Command: %s", cmd.Path)
+
+	output, _ := util.run(cmd, 4, query, MockLocal{})
+	t.Logf("Nr. of Events: %d", len(output))
+
+	for _, o := range output {
+		if o["type"] != "nvidiagpubeat" {
+			t.Errorf("event does not contain 'type' field equal to 'nvidiagpubeat'")
+		}
+	}
 }

--- a/nvidiagpubeat.template-es2x.json
+++ b/nvidiagpubeat.template-es2x.json
@@ -89,6 +89,11 @@
           "ignore_above": 1024,
           "index": "not_analyzed",
           "type": "string"
+        },
+        "type": {
+          "ignore_above": 1024,
+          "index": "not_analyzed",
+          "type": "string"
         }
       }
     }

--- a/nvidiagpubeat.template.json
+++ b/nvidiagpubeat.template.json
@@ -76,6 +76,10 @@
         "tags": {
           "ignore_above": 1024,
           "type": "keyword"
+        },
+        "type": {
+          "ignore_above": 1024,
+          "type": "keyword"
         }
       }
     }

